overlord/state/taskrunner.go

```diff
@@ -94,6 +102,7 @@ func (r *TaskRunner) run(t *Task) {
 		panic("internal error: attempted to run task with nil handler for status " + t.Status().String())
 	}
 
+	t.scheduledTime = time.Time{} // clear schedule
```

Comment:
not sure I understand the 2nd part, we are about to run it, and a cleared schedule means run immediately so that seems correct either way